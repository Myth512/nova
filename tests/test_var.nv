a := nil
print(a)
a = 1
print(a)
a += 10
print(a)

b, c := 90, 3.14

print(b, c)

tmp := b
b = c
c = tmp

print(b, c)

c = a == b

print(c)

c = a == 11

print(c)

b, c = "Hello,", " World!"

print(b + c)

d := "global"

{
	print(d)
	d := "scope1"
	print(d)
	{
		d := "scope2"
		print(d)
	}
	print(d)
}

print(d)
