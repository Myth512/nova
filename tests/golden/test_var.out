nil
1
11
90 3.14
3.14 90
false
true
Hello, World!
global
scope1
scope2
scope1
global
